lib/sparrow/fcm/v1/android.ex

```diff
@@ -296,4 +296,27 @@ defmodule Sparrow.FCM.V1.Android do
   defp add(android, key, value) do
     %{android | fields: [{key, value} | android.fields]}
   end
+
+  @spec verify(t | nil) ::
+          t | nil | {:error, :invalid_notification}
+  def verify(nil), do: nil
+
+  def verify(%{fields: [data: data]} = notification) when is_map(data) do
+    data = Enum.map(data, &verify_value/1)
+
+    case Enum.all?(data) do
+      false ->
+        {:error, :invalid_notification}
+
+      true ->
+        %{notification | fields: [{:data, Map.new(data)} | notification.fields]}
```

Comment:
What about the old `data`? This seems to duplicate entries. Maybe https://hexdocs.pm/elixir/List.html#keystore/4 ?